scripts/declarations_diff.sh

```diff
@@ -1,5 +1,44 @@
 #!/usr/bin/env bash
 
+ : <<'BASH_DOC_MODULE'
+
+#  The `declarations_diff` script
+
+The `declarations_diff` script is a text-based script that attempts to find which declarations
+have been removed and which declarations have been added in the current PR with respect to `master`.
+
+In essence, it looks at the output of `git diff origin/master...HEAD`, it extracts the lines that
+contain one of the keywords
+
+`theorem` `lemma` `inductive` `structure` `def` `class` `instance` `alias`
+
+and tries to find exact matches between a removed line and an added line
+(e.g. when moving a declaration from one file to another or restructuring within the same file).
+
+Exact matches are removed.
+Among the remaining unmatched lines, the script tries to extract the declaration ids and tries to
+find exact matches among those.
+If a declaration id is added once and removed once, then they are again considered paired and are
+not shown.
+
+If a declaration id is either only added or removed or it is added or removed more than once, then
+the script will return a count such as
+
+++--+ thmName
+
+This means that the declaration `thmName` was added 3 times and removed twice: this can happen
+with namespacing, e.g. you could see it with `++-- map_zero`.
```

Comment:
```suggestion
with namespacing, e.g. you could see it with `+++-- map_zero`.
```
With `++--` it would be silent, right?

Reply:
No, it is silent exactly with one of `+-` or `-+`.  As soon as there are more matches (or exactly one match), it reports everything